### PR TITLE
Doc 64 plugins documentation

### PR DESCRIPTION
**Pull request by tomgeudens**
**Issue**

https://graviteedevops.atlassian.net/browse/DOC-64?atlOrigin=eyJpIjoiYzFiODlkNmRjNmQzNGM1ODgzMzMyYTk4NzEwZjYzNzciLCJwIjoiamlyYS1zbGFjay1pbnQifQ

**Description**

Provided information on adding plugins to APIM

### CODE REVIEW

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+NOTE: At this time installing and updating plugins requires a restart of the component.
+
+== In a nutshell - Option II
+. Get your plugin zip-file ready.
++
+The supported plugins for Gravitee APIM can be found link:https://download.gravitee.io/#graviteeio-apim/plugins/[here, window=\"_blank\"].
+
+. Drop the plugin zip-file in a plugins-folder of your own choosing.
+
+. Modify the `gravitee.yml` configuration file of the relevant component (APIM REST API / APIM Gateway / both) to take the additional folder into account
```
Comment (marcambier) [inline pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:35]: Should we provide an example of gravitee.yml configuration file with a custom plugins folder ?

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+. Restart the component.
+
+NOTE: At this time installing and updating plugins requires a restart of the component.
+
+== Details for .ZIP installations
+=== Option I
+. Locate the plugins folder of the component
```
Comment (marcambier) [inline pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:43]: Why is there no `Get your plugin zip-file ready` on this section, like on the `In a nutshell` section ?

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# Plugins repository
+plugins:
+  path:
+    - ${gravitee.home}/plugins
+    - /customlocation/plugins
+----
+
+. Restart the component
+
+== Details for package installations (Amazon Linux, RHEL, CentOS)
```
Comment (marcambier) [inline pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:81]: I don't see any real difference in instructions for :
- in a nutshell
- zip installations
- package installations

Maybe we could some common instructions ?
(Just a suggestion, maybe it's more clear for user to distinct those cases)

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  --env gravitee_management_mongodb_uri="mongodb://gravitee-mongo:27017/gravitee-apim" \
+  --env gravitee_analytics_elasticsearch_endpoints_0="http://gravitee-elasticsearch:9200" \
+  --env gravitee_plugins_path_0=/opt/graviteeio-management-api/plugins \
+  --env gravitee_plugins_path_1=/opt/graviteeio-management-api/plugins-ext \
+  --name gravitee-apim-rest-api  \
+  --detach graviteeio/apim-management-api:latest
+
+# APIM Gateway
+docker run --publish 8082:8082 \
+  --volume /gravitee/plugins:/opt/graviteeio-gateway/plugins-ext \
```
Comment (marcambier) [inline pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:155]: should it be `/customlocation/plugins` here ?

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+*  APIM Gateway - <your installation folder>/graviteeio-apim-gateway-{{ site.products.apim._3x.version }}/config
+
++
+[source,yaml]
+----
+# Plugins repository
+plugins:
+  path:
+    - ${gravitee.home}/plugins
+    - /customlocation/plugins
```
Comment (marcambier) [inline pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:76]: Global reflexion about 'customLocation'.

A confused user could set the same plugin custom location on both rest-api and gateway side.
And will get errors running a component with plugins dedicated to another.

Maybe we have to explicit that there are 2 different folders :
- 1 for rest-api plugins
- 1 for gateway plugins

**Review by marcambier — COMMENTED** [review]
Looks good to me.
I put some suggestions/comments.

File: pages/apim/3.x/installation-guide/installation-guide-plugins.adoc (hunk 3 above)
Comment (tomgeudens) [reply pages/apim/3.x/installation-guide/installation-guide-plugins.adoc:81]: That's a good comment. The "in a nutshell" quickly describes the two basic options (put the zip file in the installation folder or put the zip file in a customer folder and modify gravitee.yml). 

Then I went through the deployment options we have in the rest of our documentation (zip, package, docker, kubernetes) describing how they are applied in each.

That is quite a bit of repetition and the differences between zip and package are minimal, agreed.

**Review by marcambier — APPROVED** [review]

**Pull request merged**